pyccel/parser/syntactic.py

```diff
@@ -260,7 +260,11 @@ def _visit_Module(self, stmt):
         return code
 
     def _visit_Expr(self, stmt):
-        return self._visit(stmt.value)
+        val = self._visit(stmt.value)
+        if isinstance(val, FunctionCall):
+            tmp_var,_ = create_variable(self._used_names)
+            val = Assign(tmp_var, val)
+        return val
```

Comment:
Can you add a few words of comment in the code? It will be good as a future reference

Reply:
What happens if the content of `Expr` does not amount to a function call, but maybe it contains some function calls and the resulting expression is not assigned to a left-hand-side variable?

Reply:
Hmm, in this case I think it will be removed. If we do:
```
3+4
```
then pyccel raises a warning and ignores this line. I think this is also the case if you do:
```
f() + g()
```
but I haven't checked this corner case

Reply:
I've just checked, they both raise the same warning
```
ERROR at shared library generation stage
pyccel:
 |warning [semantic]: tmp.py [2]| Expression with no effect has been removed (3 + 4)
```
```
ERROR at shared library generation stage
pyccel:
 |warning [semantic]: tmp.py [10]| Expression with no effect has been removed (f() + g())
```

Reply:
> Can you add a few words of comment in the code? It will be good as a future reference

Done 7a57d40

Reply:
How do we know if the expression has indeed no side effect? Shouldn't we always assign to a temporary variable and keep the expression, in order to be on the safe side?

Reply:
I agree, this is the solution that we came up with when considering expressions such as `3+4` (see #852), but function calls are much more likely to have side effects. I have ported this solution to all expressions. See e9da279

Reply:
Just to be sure... if the expression contained a function `f(x)` call which does not return anything (or that equivalently returns None), and we now assign to a temporary variable, do we still generate the correct Fortran call to a subroutine (not function)?

Reply:
We print `call f(x)`. See [here](https://github.com/pyccel/pyccel/pull/902/files#diff-9ba0c438140113a543a41e16000b939989e841d4fc6a6a4112b5c8e5d8807524R1868)

Reply:
Nice!